src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -1916,6 +1917,27 @@ public void afterSave() {
         }
     }
 
+    /**
+     * This extension intends to perform a GET request without any credentials on the provided repository URL
+     * to return the size of repository.
+     */
+    @Extension
+    public static class RepositorySizeGithubAPI extends GitToolChooser.RepositorySizeAPI {
+
+        @Override
+        public boolean isApplicableTo(String repoUrl) {
+            return repoUrl.contains("github");
```

Comment:
This is not sufficient.  GitHub Enterprise servers can have any kind of name with our without "github" in the url.  Also, there could be git repos with "github" in the name.

Reply:
Okay, I understand. So this means checking the repository url might not be enough to confirm the applicability. Would you suggest that we send some user info as well? A combination of those two could be enough to see if it is being provided by Github?